### PR TITLE
feat: Course Level Error Handling for Empty States

### PR DESCRIPTION
**Pull request by farhan-arshad-dev**
## Course Level Error Handling for Empty States
- Empty state for Home, Videos, Dates, and Discussion tabs.
- Empty state for handouts, and Announcements screen.

#### Tickets
- [LEARNER-10039](https://2u-internal.atlassian.net/browse/LEARNER-10039)
- [Issue# 339](https://github.com/openedx/openedx-app-android/issues/339)

| Home Tab | Video Tab | Dates Tab |
| ------------- | ------------- | ------------- |
|  |  |  |

| Discussion Tab | Handouts Screen | Announcements Screen |
| ------------- | ------------- | ------------- |
|  |  |  |

### CODE REVIEW

**Review by omerhabib26 — APPROVED** [review]

File: course/src/main/java/org/openedx/course/presentation/outline/CourseOutlineScreen.kt [hunk 1]
```diff
@@ -224,6 +227,10 @@ private fun CourseOutlineUI(
                 Box {
                     when (uiState) {
                         is CourseOutlineUIState.CourseData -> {
+                            if (uiState.courseStructure.blockData.isEmpty()) {
+                                NoContentScreen(noContentScreenType = NoContentScreenType.COURSE_OUTLINE)
+                                return@Surface
```
Comment (k1rill) [inline course/src/main/java/org/openedx/course/presentation/outline/CourseOutlineScreen.kt:232]: Strange to se return inside compose

**Review by k1rill — COMMENTED** [review]
Just one question

**Review by dixidroid — APPROVED** [review]

File: course/src/main/java/org/openedx/course/presentation/outline/CourseOutlineScreen.kt (hunk 1 above)
Comment (farhan-arshad-dev) [reply course/src/main/java/org/openedx/course/presentation/outline/CourseOutlineScreen.kt:232]: `NoContentScreen` is a fullscreen error, so we don't want to show other content, so if the dates are empty return the flow, after showing the No content screen.

Comment (k1rill) [reply course/src/main/java/org/openedx/course/presentation/outline/CourseOutlineScreen.kt:232]: I'm afraid that return will broke a [compose](https://developer.android.com/develop/ui/compose/mental-model) tree, can you please use else in the CourseOutlineUIState.CourseData

**Review by k1rill — CHANGES_REQUESTED** [review]
Please remove return from compose

**Review by k1rill — APPROVED** [review]

**Pull request merged**